src/oops/SOLID/singleResponsibilityPrinciple/before/EmployeeRepository.java

```diff
@@ -15,5 +21,11 @@
         Employee magda = new PartTimeEmployee("Magda Iovan", 920);
 
         return Arrays.asList(anna, billy, steve, magda);
-    }
+  }
+
+  public static void save(Employee employee) throws IOException {
```

Comment:
Nicely done, Like the abstraction of the Logger and  Persistence layer from the Repository. Just a thought though, if the underlying implementation changes, we can change the Repository itself right.

Reply:
Thanks @ishanroy.

> we can change the Repository itself right
Please refer to https://stackoverflow.com/questions/10155517/repository-pattern-vs-orm#:~:text=Repositories%20deal%20with%20Domain%2FBusiness,DTO%2C%20it%20only%20holds%20data.

Would be happy to discuss. Please let me know.